src/extension.ts

```diff
@@ -48,12 +48,21 @@ function isError(response: any): response is ErrorShape {
   );
 }
 
-export function activate(context: ExtensionContext) {
+export interface VSCodeGraphQLExtension {
+  outputChannel: OutputChannel;
+  client: LanguageClient;
+  LanguageServerCommands: typeof LSCommands;
+  LanguageServerNotifications: typeof LSNotifications;
+  LanguageServerRequests: typeof LSRequests;
+}
+
+export async function activate(
+  context: ExtensionContext,
+): Promise<VSCodeGraphQLExtension> {
   const serverModule = context.asAbsolutePath(
     join("lib/language-server", "server.js"),
   );
-
-  // Initialize language client
+  outputChannel ||= window.createOutputChannel("Apollo GraphQL");
```

Comment:
Previously, the extension often created more than one output channel and never attached to the first one. This fixes that (and makes debugging a lot easier, since there's only one channel to care about).